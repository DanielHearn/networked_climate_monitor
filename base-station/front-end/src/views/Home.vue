<template>
  <main-panel>
    <template slot="content">
      <div class="content-column">
        <h2 class="title">Introduction</h2>
        <p>
          The networked climate monitor project provides access to the recent
          and historical climate data from the sensors.
        </p>
        <p>
          This project has been designed and developed by Daniel Hearn for a
          final year project at the University of Portsmouth. The source code
          and designs for this project can be found on
          <a
            href="https://github.com/DanielHearn/networked_climate_monitor"
            class="link"
            >Github</a
          >
        </p>
        <p>
          To use the project the sensor nodes and base station must be setup,
          turned on and running their software. You must then register an
          account and login to this account. You can then access the climate
          data and manage sensors.
        </p>
        <p>
          <router-link to="/register" class="link"
            >To register click here</router-link
          >
          or
          <router-link to="/register" class="link"
            >click here to login</router-link
          >
          or
          <router-link to="/dashboard" class="link"
            >click here to view the dashboard.</router-link
          >
        </p>
      </div>
    </template>
  </main-panel>
</template>

<script>
import MainPanel from './../components/MainPanel/MainPanel.vue'

export default {
  name: 'home',
  components: {
    MainPanel
  }
}
</script>
