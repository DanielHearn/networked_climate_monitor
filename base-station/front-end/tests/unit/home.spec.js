import { shallowMount } from '@vue/test-utils'
import Home from './../../src/views/Home.vue'

describe('Home.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Home, {
      stubs: ['router-link']
    })
  })

  afterEach(() => {
    wrapper.destroy()
    jest.resetAllMocks()
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
