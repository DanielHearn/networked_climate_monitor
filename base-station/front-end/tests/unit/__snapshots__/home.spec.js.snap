// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue has the expected html structure 1`] = `
<main-panel-stub>
  <template>
    <div
      class="content-column"
    >
      <h2
        class="title"
      >
        Climate Monitor
      </h2>
       
      <h3
        class="heading"
      >
        Home
      </h3>
       
      <p>
        
        The networked climate monitor project provides access to the recent
        and historical climate data from the sensors.
      
      </p>
       
      <p>
        
        This project has been designed and developed by Daniel Hearn for a
        final year project at the University of Portsmouth. The source code
        and designs for this project can be found on
        
        <a
          class="link"
          href="https://github.com/DanielHearn/networked_climate_monitor"
        >
          Github
        </a>
      </p>
       
      <p>
        
        To use the project the sensor nodes and base station must be setup,
        turned on and running their software. You must then register an
        account and login to this account. You can then access the climate
        data and manage sensors.
      
      </p>
       
      <p>
        <router-link-stub
          class="link"
          to="/register"
        >
          To register click here
        </router-link-stub>
        
        or
        
        <router-link-stub
          class="link"
          to="/register"
        >
          click here to login
        </router-link-stub>
        
        or
        
        <router-link-stub
          class="link"
          to="/dashboard"
        >
          click here to view the dashboard.
        </router-link-stub>
      </p>
    </div>
  </template>
</main-panel-stub>
`;
